### PR TITLE
Jay - Add Controller Logic for Edit Link Permission

### PR DESCRIPTION
**Pull request by Alforoan**
# Description
A user with a 'manageAdminLinks' permission is now able to access and edit other users' links.

## Related PRS (if any):
To test this backend PR you need to checkout the [#1787](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1787) frontend PR.
…

## Main changes explained:
- Add permission logic for users to be able to access and edit other users' links.
…

## How to test:
1. check into current branch
2. do `npm install`, `npm run build` and `npm run start` to run this PR locally
3. Clear site data/cache

### CODE REVIEW

**Review by ramyaram2092 — COMMENTED** [review]
The functionality works as expected .However, seeing some exception in the backend console in addition to console statements 
image

image

**Review by pshereen — CHANGES_REQUESTED** [review]
Tested the PR.
Controller Logic for Edit Link Permission is working as per the requirement. 

But some error messages are shown at the backend console.

Screenshot 2024-01-22 201333

**Review by Renanluizssx — COMMENTED** [review]
see my review @Alforoan: https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1787

**Review by Sophie-lei — COMMENTED** [review]
Tested and left commrnt in PR1787. Please view for details.

**Review by KevHnh — APPROVED** [review]
Tested with Frontend PR and everything works as expected.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1787

**Comment by galgalinidhi** [comment]
The functionality works as expected. However, the backend console displays the following error-

Screenshot 2024-01-27 at 7 08 42 PM

**Review by Pandani07 — APPROVED** [review]
The frontend PR works as intended but I get these messages in the backend log

Screenshot 2024-01-31 at 2 06 58 PM
Screenshot 2024-01-31 at 2 07 09 PM
Screenshot 2024-01-31 at 2 07 06 PM

**Review by Pandani07 — APPROVED** [review]

**Review by SaiDeepak-NEU — APPROVED** [review]
Tested the changes for the edit link permission and everything worked as expected. I was able to see the edit link and was able to add google doc links.

Please refer to the frontend PR1787 for screenshots and additional comments.

**Review by X1aoZhang99 — CHANGES_REQUESTED** [review]
Hi Jay,

Please see my review in FE [#1787](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1787#pullrequestreview-1866835268)

**Review by mohabbasd — APPROVED** [review]
I tested this PR and it works flawlessly, good job. The screenshots are in the frontend PR review.

**Review by XiaohanMeng — COMMENTED** [review]
Hi, I have tested this PR. The detailed comment is on the FE project. Here are some errors appeared in the backend terminal while testing.
PR1787_error

File: src/controllers/userProfileController.js [hunk 1]
```diff
@@ -371,7 +373,18 @@ const userProfileController = function (UserProfile) {
...
+          req.body.requestor,
+          "putUserProfileImportantInfo"
+        )) ||
+        (await hasPermission(req.body.requestor, "manageAdminLinks"))
+      ) {
+        record.role = req.body.role;
+        record.isRehireable = req.body.isRehireable;
+        record.isActive = req.body.isActive;
+
+
```
Comment (nathanah) [inline src/controllers/userProfileController.js:387]:
```suggestion
        await hasPermission(req.body.requestor, "manageAdminLinks")
      ) {
        record.adminLinks = req.body.adminLinks;

       if (await hasPermission(req.body.requestor, "putUserProfileImportantInfo")
```
something more like this. The brackets from the if statement are causing syntax errors.

**Review by nathanah — CHANGES_REQUESTED** [review]

**Pull request closed without merging**